django/core/management/commands/runserver.py

```diff
@@ -65,6 +65,11 @@ def add_arguments(self, parser):
             action="store_true",
             help="Skip system checks.",
         )
+        parser.add_argument(
+            "--hide-unsuitable-for-production",
+            action="store_true",
+            help="Hide warning about runserver being unsuitable for production.",
+        )
```

Comment:
I think instead of adding a new argument `--hide-unsuitable-for-production`, we could perhaps stop the `--verbosity` argument being suppressed and that have the warnings when verbosity >= 1
Then `runserver --verbosity=0` can be used for people who want to ignore the warning

Reply:
I think that would defeat the purpose of the feature. We want a very explicit name to disable the warning so that people don't ship `runserver` to production.

[The forum thread has more context](https://forum.djangoproject.com/t/proposal-borrow-warning-from-werkzeug-for-runserver/32668).

Reply:
The default verbosity is 1 and so people would need to explicitly turn the verbosity to 0
Verbosity was previously not supported by runserver, so this warning will become visible to everyone, but the pattern to turn it off is well known

Reply:
Here's the scenario I'd like to avoid (and that I've been a part of, minus the setting):
1. Developers create a Dockerfile that uses `runserver --verbosity=0` because they don't want to see the warning
2. Ops are asked to "ship the app". They see the Dockerfile and are happy with it because nothing stands out as not suitable for production.
3. `runserver` gets used in production.

With an argument and/or environment variable with a very explicit name, Ops could see in the Dockerfile that there's some production related things that they need to dig into.

Reply:
We could only hide the warning when `DEBUG=True` and `runserver --verbosity=0`, so they deploy with `DEBUG=False` and they get a warning despite it being turned off by developers in their Dockerfile

Reply:
That would still be an issue in my scenario because devs used `DEBUG=True` in `settings.py` and the Ops team did not look into a very Django specific file.

Reply:
Maybe, but they've deployed with `DEBUG=True` so in this scenario they're already in trouble 
I will take this discussion back to the forum, I think the when and how should this be hidden is not obvious